File: src/server/Mimirorg.TypeLibrary/Models/Client/SelectedAttributePredefinedLibCm.cs
using Mimirorg.TypeLibrary.Enums;

namespace Mimirorg.TypeLibrary.Models.Client
{
    public class SelectedAttributePredefinedLibCm
    {
        public string Key { get; set; }
        public string Iri { get; set; }
        public ICollection<string> ContentReferences { get; set; }
        public bool IsMultiSelect { get; set; }
        public virtual Dictionary<string, bool> Values { get; set; }
        public Aspect Aspect { get; set; }

        public string Kind => nameof(SelectedAttributePredefinedLibCm);
    }
}

File: src/client/src/models/tyle/client/selectedAttributePredefinedLibCm.ts
import { Aspect } from "../enums/aspect";

export interface SelectedAttributePredefinedLibCm {
  key: string;
  iri: string;
  contentReferences: string[];
  isMultiSelect: boolean;
  values: { [key: string]: boolean };
  aspect: Aspect;
  kind: string;
}
